src/components/Dialog/ImportSongProjectDialog.vue

```diff
@@ -0,0 +1,295 @@
+import { computed, ref } from "vue";
+import { useDialogPluginComponent } from "quasar";
+import {
+  Project as UfProject,
+  EmptyProjectException,
+  IllegalFileException,
+  NotesOverlappingException,
+  supportedExtensions,
+  SupportedExtensions as UfSupportedExtensions,
+} from "@sevenc-nanashi/utaformatix-ts";
+import { useStore } from "@/store";
+import { createLogger } from "@/domain/frontend/log";
+import { ExhaustiveError } from "@/type/utility";
+import { IsEqual } from "@/type/utility";
+import { LatestProjectType } from "@/domain/project/schema";
+
+const { dialogRef, onDialogOK, onDialogCancel } = useDialogPluginComponent();
+
+const store = useStore();
+const log = createLogger("ImportExternalProjectDialog");
+
+// 受け入れる拡張子
+const acceptExtensions = computed(
+  () => supportedExtensions.map((ext) => `.${ext}`).join(",") + ",.vvproj",
+);
+
+type SupportedExtensions = UfSupportedExtensions | "vvproj";
+
+const projectNameToExtensions = [
+  ["VOICEVOX", ["vvproj"]],
+  ["MIDI (SMF)", ["mid"]],
+  ["MusicXML", ["xml", "musicxml"]],
+  ["CeVIO", ["ccs"]],
+  ["DeepVocal", ["dv"]],
+  ["OpenUtau", ["ustx"]],
+  ["Piapro Studio", ["ppsf"]],
+  ["Synthesizer V", ["s5p", "svp"]],
+  ["UtaFormatix", ["ufdata"]],
+  ["UTAU", ["ust"]],
+  ["VOCALOID", ["vpr", "vsq", "vsqx"]],
+] as const satisfies [string, SupportedExtensions[]][];
+
+// ちゃんと全部の拡張子があるかチェック
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const _: IsEqual<
+  (typeof projectNameToExtensions)[number][1][number],
+  SupportedExtensions
+> = true;
+
+// プロジェクトファイル
+const projectFile = ref<File | null>(null);
+// エラー
+const error = ref<
+  | "emptyProject"
+  | "overlapping"
+  | "parseFailed"
+  | "oldProject"
+  | "unknown"
+  | null
+>(null);
+
+// ファイルエラー
+const projectFileError = computed(() => {
+  if (!projectFile.value) {
+    return undefined;
+  }
+  if (error.value) {
+    switch (error.value) {
+      case "emptyProject":
+        return "プロジェクトが空です";
+      case "overlapping":
+        return "ノートが重なっています";
+      case "parseFailed":
+        return "ファイルの解析に失敗しました";
+      case "oldProject":
+        return "古いプロジェクトファイルです";
+      case "unknown":
+        return "不明なエラーが発生しました";
+    }
+
+    throw new ExhaustiveError(error.value);
+  } else if (project.value) {
+    if (projectTracks.value.length === 0) {
+      return "トラックがありません";
+    } else if (projectTracks.value.every((track) => track.notes.length === 0)) {
+      return "ノートがありません";
+    }
+  }
+
+  return undefined;
+});
+
+type Project =
+  | {
+      type: "utaformatix";
+      project: UfProject;
+    }
+  | {
+      type: "vvproj";
+      project: LatestProjectType;
+    };
+
+// データ
+const project = ref<Project | null>(null);
+
+const projectTracks = computed(() => {
+  if (!project.value) {
+    return [];
+  }
+  return project.value.type === "utaformatix"
+    ? project.value.project.tracks
+    : project.value.project.song.tracks;
+});
+
+// トラック
+const tracks = computed(() => {
```

Comment:
こっちは選択用なのでわかるようにしてみます
```suggestion
// 選択用のトラック
const trackOptions = computed(() => {
```